dask_geopandas/hilbert_distance.py

```diff
@@ -0,0 +1,244 @@
+import geopandas
+import numpy as np
+#from numba import jit # optional - do we want numba as a dependency for dask-geopandas?
+#ngjit = jit(nopython=True, nogil=True)
+
+def _hilbert_distance(gdf, total_bounds, p):
+
+    """
+    Calculate the hilbert distance for a GeoDataFrame based on the mid-point of 
+    the bounds for each geom and total bounds of the collection of geoms 
+    
+    Based on: https://github.com/holoviz/spatialpandas/blob/
+    9252a7aba5f8bc7a435fffa2c31018af8d92942c/spatialpandas/dask.py#L172
+    
+    Parameters
+    ----------
+    gdf          : GeoDataFrame
+    
+    total_bounds : Total bounds of GeoDataFrame
+    
+    p            : Hilbert curve parameter
+    
+    Returns
+    ---------
+    Array of hilbert distances for each geom
+    """
+    
+    if total_bounds is None:
+        total_bounds = gdf.total_bounds
+        
+    # Calculate bounds of each geom
+    bounds = gdf.bounds.to_numpy()
+    
+    # Number of dimensions
+    n = bounds.shape[1] // 2
+    # Hilbert Side len
+    side_length = 2 ** p
+    # Empty coord array
+    coords = np.zeros((bounds.shape[0], n), dtype=np.int64)
+
+    # Calculate x and y range of total bound coords - returns array
+    geom_ranges = [(total_bounds[d], total_bounds[d + n]) for d in range(n)]
+    # Calculate mid points for x and y bound coords - returns array 
+    geom_mids = [(bounds[:, d] + bounds[:, d + n]) / 2.0 for d in range(n)]
+    
+    # Transform numeric to discrete int
+    for d in range(n):
```

Comment:
Same here. `for d in range(n):` can be a bit cryptic and is not needed.

Reply:
> I will change this in relation to the above
again, I will change this here

Reply:
resolved with bc6fd49